### PR TITLE
sink(ticdc):  Add output-old-value config

### LINKED ISSUE #10916: sink(ticdc): Add output-old-value config
Add `sink.open.output-old-value`  for open protocol,  `sink.debezium.output-old-value` for debezium protocol to control weather output the old value to downstream.

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #10916

### What is changed and how it works?

this  Add `sink.open.output-old-value` for open protocol,  `sink.debezium.output-old-value` for debezium protocol, the default value is true.

#### Open Protocol Events behavior  detail 
If sink.open.output-old-value is true,  no difference 
If sink.open.output-old-value is false
- Add
No impact
```json
{
"u": [{"c1": xxx}],
"p": null,
"d": null
}
```
- Update
The  p will be null
```json
{
"u": [{"c1": xxx}],
"p": null,
"d": null
}
```
- Delete
No impact
```json
{
"d": [{"c1": xxx}],
"p": null,
"u": null
}
```
#### Debezium protocol behavior
If sink.debezium.output-old-value is true,  no difference 
If sink.debezium.output-old-value is false
- Add
No impact
- Update
The before  fields is null
```json
  {
    "before": null,
    "after": {
        "id": 111,
        "name": "scooter",
        "description": "Big 2-wheel scooter",
        "weight": 5.15
        },
    "source": {...},
    "op": "u",
    "ts_ms": 1589362330904 ,
    "transaction": null
    };
```
- Delete
No impact

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Add `sink.open.output-old-value`  for open protocol,  `sink.debezium.output-old-value` for debezium protocol to control weather output the old value to downstream.
```

### CODE REVIEW

File: cdc/api/v2/model.go [hunk 1]
```diff
@@ -717,7 +731,18 @@ func ToAPIReplicaConfig(c *config.ReplicaConfig) *ReplicaConfig {
...
+		var debeziumConfig *DebeziumConfig
+		if cloned.Sink.Debezium != nil {
+			debeziumConfig = &DebeziumConfig{
+				OutputOldValue: cloned.Sink.Debezium.OutputOldValue,
+			}
+		}
+		var openProtocolConfig *OpenProtocolConfig
+		if cloned.Sink.OpenProtocol != nil {
+			openProtocolConfig = &OpenProtocolConfig{
+				OutputOldValue: cloned.Sink.Debezium.OutputOldValue,
```
Comment (3AceShowHand) [inline cdc/api/v2/model.go:743]: cloned.Sink.Open.OutputOldValue

File: pkg/sink/codec/common/config.go [hunk 2]
```diff
@@ -76,6 +76,8 @@ type Config struct {
 
 	// for open protocol
 	OnlyOutputUpdatedColumns bool
+	// Whether old value should be excluded in the output.
+	OpenOutputOldValue bool
```
Comment (3AceShowHand) [inline pkg/sink/codec/common/config.go:80]: This should only control the `open-protocol`?

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/sink/codec/common/config.go (hunk 2 above)
Comment (sdojjy) [reply pkg/sink/codec/common/config.go:80]: yes

File: cdc/api/v2/model.go (hunk 1 above)
Comment (sdojjy) [reply cdc/api/v2/model.go]: fixed

**Review by flowbehappy — APPROVED** [review]

File: pkg/config/sink.go [hunk 3]
```diff
@@ -193,6 +193,12 @@ type SinkConfig struct {
 
 	// Debezium only. Whether schema should be excluded in the output.
 	DebeziumDisableSchema *bool `toml:"debezium-disable-schema" json:"debezium-disable-schema,omitempty"`
+
+	// OpenProtocol related configurations
+	OpenProtocol *OpenProtocolConfig `toml:"open" json:"open,omitempty"`
+
+	// DebeziumConfig related configurations
+	Debezium *DebeziumConfig `toml:"debezium" json:"debezium,omitempty"`
```
Comment (hongyunyan) [inline pkg/config/sink.go:201]: It's better to move DebeziumDisableSchema into DebeziumConfig.

**Review by hongyunyan — APPROVED** [review]
LGTM

File: pkg/config/sink.go (hunk 3 above)
Comment (sdojjy) [reply pkg/config/sink.go:201]: DebeziumDisableSchema is already published, we shouldn't break the compatibility

**Pull request merged**